unit-tests-java8/src/test/java/org/eclipse/collections/test/map/mutable/UnmodifiableMutableMapTest.java

```diff
@@ -62,133 +55,9 @@ public <K, V> MutableMap<K, V> newWithKeysValues(Object... elements)
-        assertEquals(3, map.getIfAbsentPut("3", () -> {
-            throw new AssertionError();
-        }));
-        assertEquals(this.newWithKeysValues("3", 3, "2", 2, "1", 1), map);
-
-        assertThrows(UnsupportedOperationException.class, () -> map.getIfAbsentPut("4", () -> {
-            throw new AssertionError();
-        }));
-
-        assertEquals(3, map.getIfAbsentPut("3", 4));
-        assertEquals(this.newWithKeysValues("3", 3, "2", 2, "1", 1), map);
-
-        assertThrows(UnsupportedOperationException.class, () -> map.getIfAbsentPut("4", 4));
-
-        assertEquals(3, map.getIfAbsentPutWithKey("3", key -> {
-            throw new AssertionError();
-        }));
-        assertEquals(this.newWithKeysValues("3", 3, "2", 2, "1", 1), map);
-
-        assertThrows(UnsupportedOperationException.class, () -> map.getIfAbsentPutWithKey("4", key -> {
-            throw new AssertionError();
-        }));
-
-        assertEquals(3, map.getIfAbsentPutWith("3", x -> x + 10, 4));
-        assertEquals(this.newWithKeysValues("3", 3, "2", 2, "1", 1), map);
-
-        assertThrows(UnsupportedOperationException.class, () -> map.getIfAbsentPutWith("4", x -> x + 10, 4));
-    }
-
-    @Override
-    public void MutableMapIterable_updateValue()
-    {
-        MutableMapIterable<String, Integer> map = this.newWithKeysValues("One", 1);
-        assertThrows(
-                UnsupportedOperationException.class,
-                () -> map.updateValue("One", () -> 0, ignored -> {
-                    throw new AssertionError();
-                }));
-        assertThrows(
-                UnsupportedOperationException.class,
-                () -> map.updateValue("Two", () -> 0, ignored -> {
-                    throw new AssertionError();
-                }));
-
-        assertThrows(
-                UnsupportedOperationException.class,
-                () -> map.updateValueWith("One", () -> 0, (ignored, parameter) -> {
-                    throw new AssertionError();
-                }, "test"));
-        assertThrows(
-                UnsupportedOperationException.class,
-                () -> map.updateValueWith("Two", () -> 0, (ignored, parameter) -> {
-                    throw new AssertionError();
-                }, "test"));
-    }
-
-    @Override
-    public void Map_merge()
-    {
-        MutableMapIterable<Integer, String> map = this.newWithKeysValues(1, "1", 2, "2", 3, "3");
-        assertThrows(UnsupportedOperationException.class, () -> map.merge(3, "4", (v1, v2) -> {
-            fail("Expected lambda not to be called on unmodifiable map");
-            return null;
-        }));
-        assertThrows(UnsupportedOperationException.class, () -> map.merge(4, "4", (v1, v2) -> {
-            fail("Expected lambda not to be called on unmodifiable map");
-            return null;
-        }));
-        Assert.assertEquals(this.newWithKeysValues(1, "1", 2, "2", 3, "3"), map);
-    }
```

Comment:
These assertions are pulled up and now apply to all MutableMapIterables, not just MutableMaps. This led to finding some inconsistencies in BiMaps.